src/Qir/Runtime/stdlib/include/qir_stdlib.h

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <cstdint>
+
+#ifdef _WIN32
+#define QIR_SHARED_API __declspec(dllexport)
+#else
+#define QIR_SHARED_API
+#endif
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    struct QirString;
+    struct QirArray;
+    struct QirCallable;
+    struct QirTuple;
+    struct QirBigInt;
+
+    enum PauliId : int8_t
+    {
+        PauliId_I = 0,
+        PauliId_X = 1,
+        PauliId_Z = 2,
+        PauliId_Y = 3,
+    };
+
+    typedef void (*t_CallableEntry)(QirTuple*, QirTuple*, QirTuple*);
+    typedef void (*t_CaptureCallback)(QirTuple*, int32_t);
```

Comment:
FYI: If a C file includes this header, and it is compiled with C (not C++) compiler, then I believe the compiler will complain  until we make the following change:
```suggestion
    typedef void (*t_CallableEntry)(struct QirTuple*, struct QirTuple*, struct QirTuple*);
    typedef void (*t_CaptureCallback)(struct QirTuple*, int32_t);
```
Applicable below for all the uses of identifiers
```cpp
    struct QirString;
    struct QirArray;
    struct QirCallable;
    struct QirTuple;
    struct QirBigInt;
```

Reply:
Or we can try to introduce the type aliases for the struct names:
```c
    struct QirString;    // Original structs are retained.
    struct QirArray;
    struct QirCallable;
    struct QirTuple;
    struct QirBigInt;
    
    typedef struct QirString QirString;  // Type aliases with the same name are added.
    typedef struct QirArray QirArray;
    typedef struct QirCallable QirCallable;
    typedef struct QirTuple QirTuple;
    typedef struct QirBigInt QirBigInt;
```

Reply:
Good call on the type aliases, I hadn't anticipated that. Added them!